.buildkite/scripts/generate_filebeat_pipeline.sh

```diff
@@ -0,0 +1,231 @@
+        command: "cd $BEATS_PROJECT_NAME && mage build unitTest"
+        notify:
+          - github_commit_status:
+              context: "Filebeat: MacOS ARM Unit Tests"
+        agents:
+          provider: "orka"
+          imagePrefix: "${IMAGE_MACOS_ARM}"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.xml"
+YAML
+fi
+
+if are_conditions_met_arm_tests; then
+  cat >> $pipelineName <<- YAML
+
+      - label: ":linux: ARM Ubuntu Unit Tests"
+        key: "extended-arm64-unit-test"
+        command: "cd $BEATS_PROJECT_NAME && mage build unitTest"
+        notify:
+          - github_commit_status:
+              context: "Filebeat: Unit Tests ARM"
+        agents:
+          provider: "aws"
+          imagePrefix: "${AWS_IMAGE_UBUNTU_ARM_64}"
+          instanceType: "${AWS_ARM_INSTANCE_TYPE}"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.xml"
+YAML
+fi
+
+if are_conditions_met_win_tests; then
+  cat >> $pipelineName <<- YAML
+
+  - group: "Windows Extended Testing"
+    key: "extended-tests-win"
+    steps:
+      - label: ":windows: Win 2019 Unit Tests"
+        key: "windows-extended-2019"
+        command: |
+          Set-Location -Path $BEATS_PROJECT_NAME
+          mage build unitTest
+        notify:
+          - github_commit_status:
+              context: "Filebeat: Win-2019 Unit Tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_2019}"
+          machine_type: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 200
+          disk_type: "pd-ssd"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.*"
+
+      - label: ":windows:-11 Unit Tests"
+        key: "windows-extended-11"
+        command: |
+          Set-Location -Path $BEATS_PROJECT_NAME
+          mage build unitTest
+        notify:
+          - github_commit_status:
+              context: "Filebeat: Win-11 Unit Tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_11}"
+          machine_type: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 200
+          disk_type: "pd-ssd"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.*"
+
+      - label: ":windows:-10 Unit Tests"
+        key: "windows-extended-10"
+        command: |
+          Set-Location -Path $BEATS_PROJECT_NAME
+          mage build unitTest
+        notify:
+          - github_commit_status:
+              context: "Filebeat: Win-10 Unit Tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_10}"
+          machine_type: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 200
+          disk_type: "pd-ssd"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.*"
+YAML
+fi
+
+echo "Check and add the Packaging into the pipeline"
+if are_conditions_met_packaging; then
+cat >> $pipelineName <<- YAML
+
+  - group: "Packaging"
+    key: "packaging"
+    depends_on:
+      - "mandatory-tests"
+
+    steps:
+      - label: Package pipeline
+        commands: ".buildkite/scripts/packaging/package-step.sh"
```

Comment:
This script internally creates two steps. That internally runs the `mage package,` but it also publishes this to our Docker registry, right?

I don't know if this is the right PR to solve this, but I would like to review it now if we are doing something wrong.

Reply:
fixed